src/lib/pickles_types/shifted_value.ml

```diff
@@ -25,67 +56,135 @@ open Core_kernel
    as desired.
 *)
 
-[%%versioned
-module Stable = struct
-  module V1 = struct
-    type 'f t = Shifted_value of 'f
-    [@@deriving sexp, compare, equal, yojson, hash]
+module type S = sig
```

Comment:
when I read this file I'm wondering: what does this module type represent? What can someone use it for? I think these sort of questions should be answered in documentation. For example, if this was a library facing external users, what would you write to them so that they don't need to look at the internals to use it?

Reply:
This is standard practice for OCaml: `Shifted_value.S` represents the signature that captures the 'essence' of a `Shifted_value` implementation, so that code can be written as generic over `Type1` and `Type2` defined below.